src/hooks/Api/useAuth.ts

```diff
@@ -0,0 +1,31 @@
+import { useMutation } from '@tanstack/react-query';
+import { useNavigate } from 'react-router-dom';
+
+import { PATH } from '@/routes/path';
+import { signIn } from '@/services/Auth';
+
+import { useLocalStorage } from '../useLocalStorage';
+import { SignInProps } from './type';
+
+/**
+ * 로그인 signIn API 통신 함수를 사용하는 useMutation 훅이다.
+ * - 로그인 성공 시 액세스 토큰을 로컬 스토리지에 저장한다.
+ * - 저장 후 메인 페이지(홈)으로 다이렉팅한다.
+ * @returns `mutate`, `data`
+ */
+export const useSignIn = () => {
+  const navigate = useNavigate();
+  const [, setAccessToken] = useLocalStorage('accessToken');
+
+  const { mutate, data } = useMutation({
+    mutationFn: ({ loginEmail, loginPassword }: SignInProps) => {
+      return signIn(loginEmail, loginPassword);
+    },
+    onSuccess: ({ response }) => {
+      setAccessToken(response.accessToken);
+      navigate(PATH.HOME);
+    },
+  });
+
+  return { mutate, data };
+};
```

Comment:
지금 LoginForm에서는 mutate만 사용하기 때문에 문제는 없지만
추후 useMutation의 isLoading 등의 여러 상태를 사용하는 상황이 있을 수도 있으니

useMutation 자체를 return 하는 방법을 사용해보는건 어떠신가요?!

```suggestion
  return useMutation({
    mutationFn: ({ loginEmail, loginPassword }: SignInProps) => {
      return signIn(loginEmail, loginPassword);
    },
    onSuccess: ({ response }) => {
      setAccessToken(response.accessToken);
      navigate(PATH.HOME);
    },
  });
};
```

Reply:
좋습니다! 저도 당장 필요한 것만 빼낸다는 생각에 객체 구조 분해를 사용했었는데 useMutation 반환 객체 자체를 return하는 것이 확장성이 좋을 것 같네요 👍

Reply:
혹시 useMutation을 return 하게되면 LoginForm.tsx에서 실제 사용할 때 어떻게 사용하는지 예시가 궁금합니다..!

그리구 isLoading 등을 LoginForm.tsx에서 어떻게 사용하는지도 궁금합니다.. 🙇

Reply:
useMutation 자체를 return 하게 된다고 해서 사용 방식이 바뀌는 것은 아닙니다!
이전 제 코드에서는 useMutation 반환값 내 mutate, data 만 객체 구조 분해로 뽑아서 사용한 것이고
수현님의 의견은 mutate, data 외 다른 반환값도 사용할 수 있기에 공용 커스텀 훅 useSignIn 생성 의도에 맞게 useMutation에 존재하는 모든 반환값을 사용할 수 있도록 하자는 것이었습니다.

@meowTarae 건호님이 질문주신 isLoading을 LoginForm.tsx에서 사용할 때 예시 코드를 작성해보자면 아래와 같습니다.
객체 구조 분해에서 원하는 변수명(`isSignInLoading`)으로 변경하여 사용할 수도 있습니다.
```ts
const { isLoading: isSignInLoading } = useSignIn();
```

Reply:
익준님께서 잘 설명해주신 것 같아요 감사합니다!!
isLoading 같은 경우에는 API 응답을 받기 전까지 Spinner를 그려주는 용도로 많이 사용합니다! ㅎㅎ